x/programs/cmd/simulator/src/lib.rs

```diff
@@ -137,6 +145,12 @@ impl From<u64> for Param {
     }
 }
 
+impl From<Gas> for Param {
+    fn from(val: Gas) -> Self {
+        Param::Gas(val)
+    }
+}
```

Comment:
It's tricky that way... Because then we need to implement `From<&Param> for &StringParam` which leads to return a value referenced by the local function

Reply:
... Figured that you meant `From<&Param> for StringParam`, I implemented it and I'm not actually sure if it's really clearer that way. We need to opt-out the `Key` variant because it's not at the same level as other params